packages/contracts-node/README.md

```diff
@@ -0,0 +1,71 @@
+# @uma/contracts-node
+
+This package exports all UMA smart contract artifacts specifically meant for consumption in nodejs.
+
+## Installing the package
+
+```bash
+yarn add @uma/contracts-node
+```
+
+## Importing artifacts directly
+
+To import artifacts directly into your tool of choice, the index file provides convenient functions for doing so:
+
+```js
+const { getAbi, getBytecode, getAddress } = require("@uma/contracts-node")
+
+// Gets the abi object for the Voting contract.
+// This will be needed to interact with the voting contract.
+const votingAbi = getAbi("Voting")
+
+// Gets the bytecode string for the voting contract.
+// This should usually only be required if you need to deploy a voting contract (rare).
+const votingBytecode = getBytecode("Voting")
+
+// Gets the address of the voting contract for chain id 1 (eth mainnet).
+// If there is no single canonical deployment of a particular contract on the network provided, this will fail.
+const chainId = 1
+const votingAddress = getAddress("Voting", chainId)
+```
+
+## Typescript!
+
+Typescript is fully supported for all of the above operations:
+
+```ts
+import { getAbi, getBytecode, getAddress } from "@uma/contracts-node";
+...
+```
+
+In addition to artifact support, typescript types (and sometimes factories) are available for ethers and web3.
+
+### Ethers
+
+The best support is for Ethers contract types. To construct an Ethers contract, import one of the ethers factories:
+
+```ts
+import { VotingEthers__factory, getAddress } from "@uma/contracts-node" // Factory to create ethers instance.
```

Comment:
this looks fine to me, id like to bring up potentially using types from typechain outside of the factory and contract. Don't know though if you are under the mercy of typechain, but ideally types for Events, and in lieu of that, access to the common.ts file which can be used for type inference. currently i have to do this: `"@uma/core/types/contract-types/ethers/commons"`.  This is also important if we want event types on the FE as well

Reply:
Doing blanket exports is tricky to avoid any sort of naming conflicts (and I played with namespaces, but also ran into problems there).

I looked at the Ethers contract files that typechain produced, and it appears as if the Event types don't exist outside of the contract class, itself (each contract and factory file has exactly one export).

I can export the the general event types from common under *Ethers aliases so they don't conflict with anything else.

Web3 does have some specific event types, but those are really tricky since they would need to be exported using a `*` since we don't know the events ahead of time. But since event names are certainly repeated across contracts, there will be collisions. I tried to manage this with namespaces, but ran into tons of ts compilation issues. Since nobody is using these web3 event types right now, I decided to leave that issue for another day.

Thoughts on me just adding the common.ts exports with the `Ethers` suffix?

Reply:
yea i am good with that since thats all i use currently